src/raftstore/store/worker/apply.rs

```diff
@@ -310,6 +323,29 @@ impl<'a> ApplyContext<'a> {
         });
     }
 
+    /// Stash the dirty state away for a `ApplyDelegate`, so
+    /// the context is ready to switch to apply other `ApplyDelegate`.
+    pub fn stash(&mut self, delegate: &mut ApplyDelegate) -> Stash {
+        self.commit_opt(delegate, false);
+        Stash {
+            // last cbs should not be popped, because if the ApplyContext
+            // is flushed, the callbacks can be flushed too.
+            region: self.cbs.last().map(|cbs| cbs.region.clone()),
+            exec_ctx: self.exec_ctx.take(),
+            last_applied_index: self.last_applied_index,
+        }
+    }
+
+    /// Restore the dirty state, so context can resume applying from
+    /// last stash point.
+    pub fn restore_stash(&mut self, stash: Stash) {
+        if let Some(region) = stash.region {
```

Comment:
When will this be `None`?

Reply:
When `stash` is callced before `prepare_for`.